### PR TITLE
Improve memory for HashBuilderOperator unspill

### PR DESCRIPTION
**Pull request by JunhyungSong**
Comparable changes extracted from prestodb/presto#16212

Before the fix, all unspilled data are read into memory and hold on until HashBuilderOperator is destructed.
Nullify unspilled pages allows memory to be freed.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/join/HashBuilderOperator.java [hunk 1]
```diff
@@ -556,23 +560,33 @@ private void finishLookupSourceUnspilling()
...
-            // Pages have not be unspilled yet.
+            // Pages have not been unspilled yet.
             return;
         }
 
         // Use Queue so that Pages already consumed by Index are not retained by us.
         Queue<Page> pages = new ArrayDeque<>(getDone(unspillInProgress.get()));
-        long memoryRetainedByRemainingPages = pages.stream()
+        unspillInProgress = Optional.empty();
+        long sizeOfUnSpilledPages = pages.stream()
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/join/HashBuilderOperator.java:570]: `sizeOfUnSpilledPages` -> `sizeOfUnspilledPages`

File: core/trino-main/src/main/java/io/trino/operator/join/HashBuilderOperator.java [hunk 2]
```diff
@@ -556,23 +560,33 @@ private void finishLookupSourceUnspilling()
...
+                .mapToLong(Page::getSizeInBytes)
+                .sum();
+        long retainedSizeOfUnSpilledPages = pages.stream()
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/join/HashBuilderOperator.java:573]: `retainedSizeOfUnSpilledPages` -> `retainedSizeOfUnspilledPages`

File: core/trino-main/src/main/java/io/trino/operator/join/HashBuilderOperator.java [hunk 3]
```diff
@@ -556,23 +560,33 @@ private void finishLookupSourceUnspilling()
     {
         checkState(state == State.INPUT_UNSPILLING);
         if (!unspillInProgress.get().isDone()) {
-            // Pages have not be unspilled yet.
+            // Pages have not been unspilled yet.
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/join/HashBuilderOperator.java:563]: nit: this could be separate commit

**Review by sopel39 — APPROVED** [review]
lgtm % small comments

**Review by arhimondr — APPROVED** [review]
LGTM % @sopel39 's comments

**Comment by sopel39** [comment]
@JunhyungSong could you sign CLA please?

**Comment by JunhyungSong** [comment]
> @JunhyungSong could you sign CLA please?

I have done that. can you check that, please?

**Comment by martint** [comment]
@cla-bot check
